tests/server/api/v1/test_datasets.py

```diff
@@ -1012,9 +1014,15 @@ def test_create_dataset_records_with_nonexistent_dataset_id(client: TestClient,
     assert db.query(Response).count() == 0
 
 
-def test_publish_dataset(client: TestClient, db: Session, admin_auth_header: dict):
+@pytest.mark.skipif(condition=not is_running_elasticsearch(), reason="Test only running with elasticsearch backend")
+def test_publish_dataset(
+    client: TestClient,
+    db: Session,
+    elasticsearch: Elasticsearch,
```

Comment:
If we change this to `search_engine` on handlers maybe we should change it here too:

```suggestion
    search_engine: Elasticsearch,
```

Remember to change it for the rests of tests.

Reply:
This is not the search engine, is just the official elasticsearch client used to check test results.